python/ai_service.py

```
@@ -207,6 +231,31 @@ def speech_to_text(self, audio_data: bytes, language: str) -> str:
+
+        try:
+            file_name = "audio.wav"
+            whisper_language = language if language else None
+            response = self.openai_client.audio.transcriptions.create(
+                model=self.whisper_model,
+                file=(file_name, audio_data),
+                language=whisper_language,
+            )
+            transcription = response.text.strip()
+            logger.info("Whisper transcription succeeded")
+            return transcription
```

Ruff (0.14.6)

Consider moving this statement to an `else` block

(TRY300)